calendar.js

```diff
@@ -3,107 +3,122 @@ function myTable() {
+    $("#confirm").click({param1: parseInt( $(this).index() - 1)}, confirmModal);
 }
 
 
 //function to create the skeleton of the calendar
 function createTable() {
-    var table = document.createElement('table');
+    var table, thElements, k, th, i, tr, td, j;
+    table = document.createElement('table');
     table.id = "calendar-id";
-    var thElements = ['Hour of day', 'Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
-
-    for (var k = 0; k < 8; k++) {
-        var th = document.createElement('th');
-        th.innerText = thElements[k];
+    thElements = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+    th = document.createElement('th');
+    th.innerText = 'Hour of day';
+    table.appendChild(th);
+    thElements.forEach(function(day, i) {
+        th = document.createElement('th');
+        th.id = i + day;
+        th.innerText = day;
         table.appendChild(th);
-    }
-    for (var i = 0; i < 24; i++) {
-        var tr = document.createElement('tr');
+    });
+
+    for (i = 0; i < 24; i++) {
+        tr = document.createElement('tr');
         tr.id = i;
         table.appendChild(tr);
-        var td = document.createElement('td');
+        td = document.createElement('td');
         td.innerText = i + ":00";
         tr.appendChild(td);
-
-        for (var j = 1; j < 8; j++) {
-            var td = document.createElement('td');
-            updateCellWithNoAppointment(td);
+        thElements.forEach(function(day, j) {
+            td = document.createElement('td');
+            td.id = day + i;
             tr.appendChild(td);
-        }
+        });
     }
     return table;
 }
 //function which makes an ajax call to get the values from the json file which contains the hours for which appointments have been booked
 function getScheduleAndUpdateTable(table) {
-    var time = [];
     $.ajax({
-        url: './calendar.json',
+        url: '/getjsonresponse',
         type: 'get',
         cache: false,
         success: function (data) {
-            $.each(data, function (i, item) {
-                time[i] = item.time;
-            });
-            updateTable(table, time);
-        }
+            updateTable(table,data);
+        }   
     });
 }
 //function to update the table according to the booked appointments
-function updateTable(table, time) {
-    for (var i = 0; i < 24; i++) {
-        var tr = document.getElementById(i);
-        for (var j = 0; j < 7; j++) {
-            var td = tr.getElementsByTagName('td');
-
-            if (time[j].length == 0) {
-                updateCellWithNoAppointment(td[j + 1]);
+function updateTable(table, data) {
+    var eventName, startRow, startTimeDay, locationName, startTimeHour, endTimeHour;
+    $.each(data, function (i, item) {
+        eventName = item.eventName;
+        locationName = item.locationName;
+        startTimeDay = (new Date(item.startTime * 1000)).toLocaleString('en-US', {weekday: 'short'});
```

Comment:
code is doing too much, please break it down into a separate function like `formatDateToEpoch`